File: Recursion/lecture 4/Fraz_recursion_4.java
import java.util.*;
import java.lang.*;

class Power {
    public static long Pow(int X, int N) {
        if (N == 0)
            return 1;
        long temp = Pow(X, N / 2);
        if (N % 2 == 1)
            return temp * temp * X;
        return temp * temp;
    }

    public static void main(String[] args) {
        int x = 0;
        Scanner sc = new Scanner(System.in);
        x = sc.nextInt();
        int n = 0;
        n = sc.nextInt();
        System.out.println(Pow(x, n));
    }

}

File: Recursion/lecture 4/Fraz_recursion_4.py
def Pow(X,N):
    if N == 0:
        return 1    
    temp = Pow(X,N//2)
    if(N%2 == 1):
        return temp* temp*X
    return temp * temp

x = int(input())
n = int(input())
print(Pow(x,n))